### ISSUE TITLE
ObservableOnWrite, ObservableOnChange

### ISSUE THREAD

**Issue by JeffreySarnoff** [JuliaGizmos/Observables.jl#5]
At present, your Observable values report on every assignment.  There are plenty of use cases where that all that activity could be an encumbrance.   A second serving of observing, one that reports on changed value (an assignment of a value that is not equal to the current value held) should be considered.  One might use `isapprox` with Floats.  The third kind would report on change of value which exceeds an initialization time safe range.

**Comment by ianshmean** [comment]
Are adding these options on the horizon? If not is there a recommended way to implement ObservableOnChange-like behavior?

**Comment by travigd** [comment]
Borrowing from Rx terminology, here's a simple `distinct_until_changed`. I make no claims that this is the best way to do this, but, it is one way.

```julia
function distinct_until_changed(obs::Observable{T}) where {T}
    mapped = Observable{T}(obs[])
    on(obs) do next
        if next != mapped[]
            mapped[] = next
        end
    end
    return mapped
end
```

```julia
julia> obs = Observable("foo");

julia> obs2 = distinct_until_changed(obs);

julia> on(obs2) do value; println("obs2 got $value"); end;

julia> obs[] = "foo";

julia> obs[] = "bar";
obs2 got bar
```

**Comment by timholy** [comment]
The update-on-changed-value was fixed by #68. There's also `throttle`. If there are still missing cases please file another issue.
